mailpoet/lib/Subscription/AdminUserSubscription.php

```diff
@@ -0,0 +1,171 @@
+    $this->loggerFactory = $loggerFactory;
+  }
+
+  /**
+   * Set up hooks for the Add New User form in WordPress admin
+   */
+  public function setupHooks(): void {
+    // Set up hooks for the Add New User form
+    $this->wp->addAction('user_new_form', [$this, 'displaySubscriberStatusField']);
+    
+    // Use the lower priority filter instead of an action - this is simpler and more reliable
+    $this->wp->addFilter('mailpoet_subscriber_data_before_save', [$this, 'modifySubscriberData'], 10, 1);
+    
+    // Add action to send confirmation email after user registration is complete
+    $this->wp->addAction('user_register', [$this, 'maybeSendConfirmationEmail'], 30, 1);
+  }
+
+  /**
+   * Display the subscriber status field on the Add New User form
+   *
+   * @param string $type The form context, 'add-new-user' for single site and network admin
+   */
+  public function displaySubscriberStatusField($type): void {
+    // According to WordPress docs, the parameter is 'add-new-user' for single site and network admin
+    if ($type !== 'add-new-user') {
+      return;
+    }
+
+    $confirmationEnabled = (bool)$this->settings->get('signup_confirmation.enabled', false);
+    $defaultStatus = $confirmationEnabled ?
+      SubscriberEntity::STATUS_UNCONFIRMED :
+      SubscriberEntity::STATUS_UNSUBSCRIBED;
+
+    // Include the template file
+    include __DIR__ . '/../../views/subscription/admin_user_status_field.php';
+  }
+
+  /**
+   * Modify subscriber data before save during user creation
+   *
+   * @param array $data The subscriber data
+   * @return array Modified subscriber data
+   */
+  public function modifySubscriberData($data): array {
+    // Only process during user creation in admin
+    if (!$this->isAdminUserCreation()) {
+      return $data;
+    }
+    
+    // Check if our field was submitted
+    if (!isset($_POST['mailpoet_subscriber_status'])) {
+      return $data;
+    }
+
+    $status = sanitize_text_field($_POST['mailpoet_subscriber_status']);
+
+    // Validate the status value
+    $validStatuses = [
+      SubscriberEntity::STATUS_SUBSCRIBED,
+      SubscriberEntity::STATUS_UNCONFIRMED,
+      SubscriberEntity::STATUS_UNSUBSCRIBED,
+    ];
+
+    if (!in_array($status, $validStatuses)) {
+      return $data;
+    }
+
+    // Update the subscriber data
+    $data['status'] = $status;
+    $data['source'] = 'administrator';
+
+    return $data;
+  }
+
+  /**
+   * Send confirmation email if needed after user registration
+   *
+   * @param int $userId The WordPress user ID
+   */
+  public function maybeSendConfirmationEmail(int $userId): void {
+    // Only process during user creation in admin
+    if (!$this->isAdminUserCreation()) {
+      return;
+    }
+    
+    // Check if our field was submitted and is set to unconfirmed
+    if (
+        !isset($_POST['mailpoet_subscriber_status']) || 
+        $_POST['mailpoet_subscriber_status'] !== SubscriberEntity::STATUS_UNCONFIRMED
```

Comment:
I am surprised this does not throw phpcs errors like "you need to use sanitize_text_field" and stuff. usage is safe though 👍